design/tracing-design.md

```diff
@@ -0,0 +1,170 @@
+      # literal is a static custom tag value.
+      literal: 'this is literal'
+    - tagName: environment
+      # environment indicates that the label value is obtained from the environment variable.
+      environment: HOST
+    - tagName: requestHeaderName
+      # requestHeaderName indicates which request header the label value is obtained from.
+      requestHeaderName: ':path'
+  # extensionService Identifies the extension service defining the openTelemetry collector service.
+  # formatted as <namespace>/<name>.    
+  extensionService: projectcontour/otel-collector
+```
+
+### HCM Configuration
+
+In the presence of the `tracing` block above, all HTTP connection managers (HCM) will be configured for tracing.
+
+A Tracing HCM will include the following as an example:
+
+```go
+...
+Tracing: &http.HttpConnectionManager_Tracing{
+            OverallSampling: &envoy_type.Percent{
+                Value: 100,
+            },
+            MaxPathTagLength: wrapperspb.UInt32(256),
+            CustomTags:       []*envoy_trace_v3.CustomTag{
+                {
+                    Tag: "literal",
+                    Type: &envoy_trace_v3.CustomTag_Literal_{
+                        Literal: &envoy_trace_v3.CustomTag_Literal{
+                            Value: "this is literal",
+                        },
+                    },
+                },
+                {
+                    Tag: "environment",
+                    Type: &envoy_trace_v3.CustomTag_Environment_{
+                        Environment: &envoy_trace_v3.CustomTag_Environment{
+                            Name: "HOST",
+                        },
+                    },
+                },
+                {
+                    Tag: "requestHeaderName",
+                    Type: &envoy_trace_v3.CustomTag_RequestHeader{
+                        RequestHeader: &envoy_trace_v3.CustomTag_Header{
+                        Name:         ":path",
+                        },
+                    },
+                },
+            },
+            Provider: &envoy_config_trace_v3.Tracing_Http{
+                Name: "envoy.tracers.opentelemetry",
+                ConfigType: &envoy_config_trace_v3.Tracing_Http_TypedConfig{
+                    TypedConfig: protobuf.MustMarshalAny(&envoy_config_trace_v3.OpenTelemetryConfig{
+                        GrpcService: &envoy_core_v3.GrpcService{
+                            TargetSpecifier: &envoy_core_v3.GrpcService_EnvoyGrpc_{
+                                EnvoyGrpc: &envoy_core_v3.GrpcService_EnvoyGrpc{
+                                    ClusterName: "extension/projectcontour/otle-collector",
+                                    Authority:   "extension/projectcontour/otle-collector",
+                                },
+                            },
+                            Timeout: durationpb.New(time.Millisecond * 50),
+                        },
+                        ServiceName: "contour-envoy-otel.otel-collector",
```

Comment:
This should not be the name of the collector service but a user defined service name via the configuration. The default could be the `ingressClassName` of Contour.

Reply:
hm yeah looks like something that should possibly be configurable so traces can be identified how the user want to, will need to think about this more, the idea of having `contour-<ingressclassname>` is interesting for a default

Reply:
Yes, this is a good idea, serviceName becomes configurable, default value is `contour-<ingressClassName>`

Reply:
The ingressClassName setting supported by contour is an array type. If the configured ingressClassName has multiple values, I don't think of a good way to deal with it, so I want to change the default value from `contour-<ingressClassName>` to `contour`.

Reply:
Yep that makes sense @yangyy93